vector/src/main/java/im/vector/app/features/notifications/NotificationDrawerManager.kt

```diff
@@ -50,42 +48,24 @@ class NotificationDrawerManager @Inject constructor(private val context: Context
     private val handlerThread: HandlerThread = HandlerThread("NotificationDrawerManager", Thread.MIN_PRIORITY)
     private var backgroundHandler: Handler
 
-    init {
-        handlerThread.start()
-        backgroundHandler = Handler(handlerThread.looper)
-    }
-
-    /**
-     * The notifiable events to render
-     * this is our source of truth for notifications, any changes to this list will be rendered as notifications
-     * when events are removed the previously rendered notifications will be cancelled
-     * when adding or updating, the notifications will be notified
-     *
-     * Events are unique by their properties, we should be careful not to insert multiple events with the same event-id
-     */
-    private val queuedEvents = loadEventInfo()
+    // TODO Multi-session: this will have to be improved
+    private val currentSession: Session?
+        get() = activeSessionDataSource.currentValue?.orNull()
 
     /**
-     * The last known rendered notifiable events
-     * we keep track of them in order to know which events have been removed from the eventList
-     * allowing us to cancel any notifications previous displayed by now removed events
+     * Lazily initializes the NotificationState as we rely on having a current session in order to fetch the persisted queue of events
      */
-    private var renderedEvents = emptyList<ProcessedEvent<NotifiableEvent>>()
+    private val notificationState by lazy { NotificationState.createInitialNotificationState(context, currentSession) }
```

Comment:
not sure if the preloading was working in the past :thinking:  we have to wait for an active session before we can query for the previously persisted events